From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sandeep Patil <sspatil@google.com>
Date: Thu, 8 Aug 2019 08:00:53 -0700
Subject: ANDROID: staging: ion: Move ion heaps into their own directory

This is preparatory work for ION heaps to be kernel modules, so
they stay in their own directory inside ION. The heap modules are
planned to be *overridable*, so it makes sense to isolate them
in source to explicitly draw the line between what is a kernel API.

Bug: 133508579
Test: ion-unit-tests

Change-Id: Iefc4e22d186139832f54b9cfc629383fb6698fc6
Signed-off-by: Sandeep Patil <sspatil@google.com>
---
 drivers/staging/android/ion/Kconfig               | 15 +--------------
 drivers/staging/android/ion/Makefile              |  5 ++---
 drivers/staging/android/ion/heaps/Kconfig         | 15 +++++++++++++++
 drivers/staging/android/ion/heaps/Makefile        |  3 +++
 .../android/ion/{ => heaps}/ion_cma_heap.c        |  2 +-
 .../android/ion/{ => heaps}/ion_page_pool.c       |  0
 .../android/ion/{ => heaps}/ion_page_pool.h       |  0
 .../android/ion/{ => heaps}/ion_system_heap.c     |  2 +-
 8 files changed, 23 insertions(+), 19 deletions(-)
 create mode 100644 drivers/staging/android/ion/heaps/Kconfig
 create mode 100644 drivers/staging/android/ion/heaps/Makefile
 rename drivers/staging/android/ion/{ => heaps}/ion_cma_heap.c (99%)
 rename drivers/staging/android/ion/{ => heaps}/ion_page_pool.c (100%)
 rename drivers/staging/android/ion/{ => heaps}/ion_page_pool.h (100%)
 rename drivers/staging/android/ion/{ => heaps}/ion_system_heap.c (99%)

diff --git a/drivers/staging/android/ion/Kconfig b/drivers/staging/android/ion/Kconfig
index 989fe84a9f9d..7b7da979991e 100644
--- a/drivers/staging/android/ion/Kconfig
+++ b/drivers/staging/android/ion/Kconfig
@@ -11,17 +11,4 @@ menuconfig ION
 	  If you're not using Android its probably safe to
 	  say N here.
 
-config ION_SYSTEM_HEAP
-	bool "Ion system heap"
-	depends on ION
-	help
-	  Choose this option to enable the Ion system heap. The system heap
-	  is backed by pages from the buddy allocator. If in doubt, say Y.
-
-config ION_CMA_HEAP
-	bool "Ion CMA heap support"
-	depends on ION && DMA_CMA
-	help
-	  Choose this option to enable CMA heaps with Ion. This heap is backed
-	  by the Contiguous Memory Allocator (CMA). If your system has these
-	  regions, you should say Y here.
+source "drivers/staging/android/ion/heaps/Kconfig"
diff --git a/drivers/staging/android/ion/Makefile b/drivers/staging/android/ion/Makefile
index b6fab2816781..1f7704e02fb0 100644
--- a/drivers/staging/android/ion/Makefile
+++ b/drivers/staging/android/ion/Makefile
@@ -1,4 +1,3 @@
 # SPDX-License-Identifier: GPL-2.0
-obj-$(CONFIG_ION) += ion.o ion_buffer.o ion_dma_buf.o ion_heap.o
-obj-$(CONFIG_ION_SYSTEM_HEAP) += ion_system_heap.o ion_page_pool.o
-obj-$(CONFIG_ION_CMA_HEAP) += ion_cma_heap.o
+obj-$(CONFIG_ION)	+= ion.o ion_buffer.o ion_dma_buf.o ion_heap.o
+obj-y			+= heaps/
diff --git a/drivers/staging/android/ion/heaps/Kconfig b/drivers/staging/android/ion/heaps/Kconfig
new file mode 100644
index 000000000000..7affa91ecabb
--- /dev/null
+++ b/drivers/staging/android/ion/heaps/Kconfig
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: GPL-2.0
+config ION_SYSTEM_HEAP
+	bool "Ion system heap"
+	depends on ION
+	help
+	  Choose this option to enable the Ion system heap. The system heap
+	  is backed by pages from the buddy allocator. If in doubt, say Y.
+
+config ION_CMA_HEAP
+	bool "Ion CMA heap support"
+	depends on ION && DMA_CMA
+	help
+	  Choose this option to enable CMA heaps with Ion. This heap is backed
+	  by the Contiguous Memory Allocator (CMA). If your system has these
+	  regions, you should say Y here.
diff --git a/drivers/staging/android/ion/heaps/Makefile b/drivers/staging/android/ion/heaps/Makefile
new file mode 100644
index 000000000000..127912629f0b
--- /dev/null
+++ b/drivers/staging/android/ion/heaps/Makefile
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: GPL-2.0
+obj-$(CONFIG_ION_SYSTEM_HEAP) += ion_system_heap.o ion_page_pool.o
+obj-$(CONFIG_ION_CMA_HEAP) += ion_cma_heap.o
diff --git a/drivers/staging/android/ion/ion_cma_heap.c b/drivers/staging/android/ion/heaps/ion_cma_heap.c
similarity index 99%
rename from drivers/staging/android/ion/ion_cma_heap.c
rename to drivers/staging/android/ion/heaps/ion_cma_heap.c
index bf65e67ef9d8..0872e5d630f8 100644
--- a/drivers/staging/android/ion/ion_cma_heap.c
+++ b/drivers/staging/android/ion/heaps/ion_cma_heap.c
@@ -14,7 +14,7 @@
 #include <linux/scatterlist.h>
 #include <linux/highmem.h>
 
-#include "ion.h"
+#include "../ion.h"
 
 struct ion_cma_heap {
 	struct ion_heap heap;
diff --git a/drivers/staging/android/ion/ion_page_pool.c b/drivers/staging/android/ion/heaps/ion_page_pool.c
similarity index 100%
rename from drivers/staging/android/ion/ion_page_pool.c
rename to drivers/staging/android/ion/heaps/ion_page_pool.c
diff --git a/drivers/staging/android/ion/ion_page_pool.h b/drivers/staging/android/ion/heaps/ion_page_pool.h
similarity index 100%
rename from drivers/staging/android/ion/ion_page_pool.h
rename to drivers/staging/android/ion/heaps/ion_page_pool.h
diff --git a/drivers/staging/android/ion/ion_system_heap.c b/drivers/staging/android/ion/heaps/ion_system_heap.c
similarity index 99%
rename from drivers/staging/android/ion/ion_system_heap.c
rename to drivers/staging/android/ion/heaps/ion_system_heap.c
index d1cb9e69399e..8e26016082d4 100644
--- a/drivers/staging/android/ion/ion_system_heap.c
+++ b/drivers/staging/android/ion/heaps/ion_system_heap.c
@@ -14,7 +14,7 @@
 #include <linux/slab.h>
 #include <linux/vmalloc.h>
 
-#include "ion.h"
+#include "../ion.h"
 #include "ion_page_pool.h"
 
 #define NUM_ORDERS ARRAY_SIZE(orders)
